File: mdiMain.cs
using System;
using System.Windows.Forms;

namespace BossAdmin
{
    public partial class mdiMain
    {
        public mdiMain()
        {
            InitializeComponent();
        }

        private void NotificationsToolStripMenuItem_Click(object sender, EventArgs e)
        {

        }

        private void MilestonesToolStripMenuItem_Click(object sender, EventArgs e)
        {
            var oForm = new Milestones();
            OpenMDIChild(oForm);

        }
        private void VPOReasonsToolStripMenuItem_Click(object sender, EventArgs e)
        {
           VPOReasons oForm = new VPOReasons();
           OpenMDIChild(oForm);

        }


        /// <summary>
        /// 
        /// </summary>
        /// <param name="oForm"></param>
        public void OpenMDIChild(Form oForm)
        {
            // this code adds the form to the windows list and allows multiples of the same window
            // if we open a window from within another window, we should use this module
            // if we are setting captions based on criteria, we can set them before the form is passwed3
            //var newMDIChild = new Form();
            Form newMDIChild = new Form();
            newMDIChild=oForm;
            newMDIChild.MdiParent=this;
            newMDIChild.Show();
        }
        private void mdiMain_Load(object sender, EventArgs e)
        {
            string sVersion = "";
            if (string.IsNullOrEmpty(modGlobals.gsDatabase))
            {
                Application.Exit();
            }
            else // mrb 10/21/14
            {
                sVersion=My.MyProject.Application.Info.Version.ToString();
                Text="BOSSAdmin V "+sVersion;
                MenusShow();
            }
        }

        private void MenusShow()
        {
            bool bIsCostAnalysis;
            bIsCostAnalysis=modGlobals.gsDatabase=="HCHDataQA";
            // add security 9/15/16 mrb
            TakeoffTablesToolStripMenuItem.Visible=bIsCostAnalysis==false&modCommon.HasViewSecurity("mnuTakeOffTable", "Menu");
            PlanEstimateTablesToolStripMenuItem.Visible=bIsCostAnalysis==false&modCommon.HasViewSecurity("mnuPlanEstimateTable", "Menu");
            MaintenanceToolStripMenuItem.Visible=bIsCostAnalysis==false;
            tsMnuItemCostAnalysis.Visible=bIsCostAnalysis;
            ViewEstimateToolStripMenuItem.Visible=bIsCostAnalysis==true;
            UsersToolStripMenuItem.Visible=modCommon.HasViewSecurity("mnuBAUserAdmin", "Menu");
            PositionsToolStripMenuItem.Visible=modCommon.HasViewSecurity("mnuBAPositions", "Menu");
            DivisionsToolstripMenu.Visible=modCommon.HasViewSecurity("mnuBAAddEditDivision", "Menu");
            ProjectsToolStripMenuItem.Visible=modCommon.HasViewSecurity("mnuBAProjectMaintenance", "Menu");
            SecurityToolStripMenuItem.Visible=modCommon.HasViewSecurity("mnuBASecurity", "Menu");
            VPOReasonsToolStripMenuItem.Visible=modCommon.HasViewSecurity("mnuBAVPOReasons", "Menu");
            MilestonesToolStripMenuItem.Visible=modCommon.HasViewSecurity("mnuBAMilestones", "Menu");
            // HelpToolStripMenuItem.Visible = False 'mrb 6/22/22
            tsMnuItemCostAnalysisReports.Visible=false; // mrb 6/22/22
        }

        private void PositionsToolStripMenuItem_Click(object sender, EventArgs e)
        {
            Positions oForm = new Positions();
            OpenMDIChild(oForm);
        }

        private void UsersToolStripMenuItem_Click(object sender, EventArgs e)
        {
            var oForm = new UserMaintenance();
            OpenMDIChild(oForm);
        }

        private void ToolStripMenuItem2_Click(object sender, EventArgs e)
        {
            var oForm = new Divisions();
            OpenMDIChild(oForm);
        }

        private void ProjectsToolStripMenuItem_Click(object sender, EventArgs e)
        {
            var oForm = new ProjectMaintenance();
            OpenMDIChild(oForm);
        }

        private void PlanEstimateTablesToolStripMenuItem_Click(object sender, EventArgs e)
        {
            var oForm = new PlanEstimateTablesMaintenance();
            OpenMDIChild(oForm);
        }

        private void tsMnuItemCostAnalysisSearch_Click(object sender, EventArgs e)
        {
            var oForm = new CostAnalysisSearch();
            OpenMDIChild(oForm);
        }

        private void ViewEstimateToolStripMenuItem_Click(object sender, EventArgs e)
        {
            var oForm = new EstimateView();
            OpenMDIChild(oForm);
        }
        public void ViewEstimateForm(string sJobID)
        {
            var oForm = new EstimateView();
            oForm.msJobID=sJobID;
            oForm.mbLockJobs=false;
            OpenMDIChild(oForm);
        }

        private void TakeoffTablesToolStripMenuItem_Click(object sender, EventArgs e)
        {
            var oForm = new PlanRoomsMaintenance();
            OpenMDIChild(oForm);
        }

        private void VendorCostToolStripMenuItem_Click(object sender, EventArgs e)
        {
            var oForm = new VendorCosts();
            OpenMDIChild(oForm);
        }

        private void MaintenanceToolStripMenuItem_Click(object sender, EventArgs e)
        {

        }
    }
}

File: mdiMain.vb
Public Class mdiMain

    Private Sub NotificationsToolStripMenuItem_Click(sender As System.Object, e As System.EventArgs) Handles NotificationsToolStripMenuItem.Click

    End Sub

    Private Sub MilestonesToolStripMenuItem_Click(sender As System.Object, e As System.EventArgs) Handles MilestonesToolStripMenuItem.Click
        Dim oForm As New Milestones
        OpenMDIChild(oForm)

    End Sub
    Private Sub VPOReasonsToolStripMenuItem_Click(sender As System.Object, e As System.EventArgs) Handles VPOReasonsToolStripMenuItem.Click
        Dim oForm As New VPOReasons
        OpenMDIChild(oForm)

    End Sub
    Public Sub OpenMDIChild(ByRef oForm As Form)
        'this code adds the form to the windows list and allows multiples of the same window
        'if we open a window from within another window, we should use this module
        ' if we are setting captions based on criteria, we can set them before the form is passwed3
        Dim newMDIChild As New Form
        newMDIChild = oForm
        newMDIChild.mdiparent = Me
        newMDIChild.Show()
    End Sub
    Private Sub mdiMain_Load(sender As Object, e As System.EventArgs) Handles Me.Load
        Dim sVersion As String = ""
        If gsDatabase = "" Then
            Application.Exit()
        Else 'mrb 10/21/14
            sVersion = My.Application.Info.Version.ToString
            Me.Text = "BOSSAdmin V " & sVersion
            MenusShow()
        End If
    End Sub

    Private Sub MenusShow()
        Dim bIsCostAnalysis As Boolean
        bIsCostAnalysis = gsDatabase = "HCHDataQA"
        'add security 9/15/16 mrb
        TakeoffTablesToolStripMenuItem.Visible = bIsCostAnalysis = False And HasViewSecurity("mnuTakeOffTable", "Menu")
        PlanEstimateTablesToolStripMenuItem.Visible = bIsCostAnalysis = False And HasViewSecurity("mnuPlanEstimateTable", "Menu")
        MaintenanceToolStripMenuItem.Visible = bIsCostAnalysis = False
        tsMnuItemCostAnalysis.Visible = bIsCostAnalysis
        ViewEstimateToolStripMenuItem.Visible = bIsCostAnalysis = True
        UsersToolStripMenuItem.Visible = HasViewSecurity("mnuBAUserAdmin", "Menu")
        PositionsToolStripMenuItem.Visible = HasViewSecurity("mnuBAPositions", "Menu")
        DivisionsToolstripMenu.Visible = HasViewSecurity("mnuBAAddEditDivision", "Menu")
        ProjectsToolStripMenuItem.Visible = HasViewSecurity("mnuBAProjectMaintenance", "Menu")
        SecurityToolStripMenuItem.Visible = HasViewSecurity("mnuBASecurity", "Menu")
        VPOReasonsToolStripMenuItem.Visible = HasViewSecurity("mnuBAVPOReasons", "Menu")
        MilestonesToolStripMenuItem.Visible = HasViewSecurity("mnuBAMilestones", "Menu")
        ' HelpToolStripMenuItem.Visible = False 'mrb 6/22/22
        tsMnuItemCostAnalysisReports.Visible = False 'mrb 6/22/22
    End Sub

    Private Sub PositionsToolStripMenuItem_Click(sender As System.Object, e As System.EventArgs) Handles PositionsToolStripMenuItem.Click
        Dim oForm As New Positions
        OpenMDIChild(oForm)
    End Sub

    Private Sub UsersToolStripMenuItem_Click(sender As System.Object, e As System.EventArgs) Handles UsersToolStripMenuItem.Click
        Dim oForm As New UserMaintenance
        OpenMDIChild(oForm)
    End Sub

    Private Sub ToolStripMenuItem2_Click(sender As System.Object, e As System.EventArgs) Handles DivisionsToolstripMenu.Click
        Dim oForm As New Divisions
        OpenMDIChild(oForm)
    End Sub

    Private Sub ProjectsToolStripMenuItem_Click(sender As System.Object, e As System.EventArgs) Handles ProjectsToolStripMenuItem.Click
        Dim oForm As New ProjectMaintenance
        OpenMDIChild(oForm)
    End Sub

    Private Sub PlanEstimateTablesToolStripMenuItem_Click(sender As Object, e As EventArgs) Handles PlanEstimateTablesToolStripMenuItem.Click
        Dim oForm As New PlanEstimateTablesMaintenance
        OpenMDIChild(oForm)
    End Sub

    Private Sub tsMnuItemCostAnalysisSearch_Click(sender As Object, e As EventArgs) Handles tsMnuItemCostAnalysisSearch.Click
        Dim oForm As New CostAnalysisSearch
        OpenMDIChild(oForm)
    End Sub

    Private Sub ViewEstimateToolStripMenuItem_Click(sender As Object, e As EventArgs) Handles ViewEstimateToolStripMenuItem.Click
        Dim oForm As New EstimateView
        OpenMDIChild(oForm)
    End Sub
    Public Sub ViewEstimateForm(sJobID As String)
        Dim oForm As New EstimateView
        oForm.msJobID = sJobID
        oForm.mbLockJobs = False
        OpenMDIChild(oForm)
    End Sub

    Private Sub TakeoffTablesToolStripMenuItem_Click(sender As Object, e As EventArgs) Handles TakeoffTablesToolStripMenuItem.Click
        Dim oForm As New PlanRoomsMaintenance
        OpenMDIChild(oForm)
    End Sub

    Private Sub VendorCostToolStripMenuItem_Click(sender As Object, e As EventArgs) Handles VendorCostToolStripMenuItem.Click
        Dim oForm As New VendorCosts
        OpenMDIChild(oForm)
    End Sub

    Private Sub MaintenanceToolStripMenuItem_Click(sender As Object, e As EventArgs) Handles MaintenanceToolStripMenuItem.Click

    End Sub
End Class

